### ISSUE TITLE
sonarjs/cognitive-complexity strange typings

### ISSUE THREAD

**Issue by datismoa** [SonarSource/eslint-plugin-sonarjs#441]
[That](https://github.com/SonarSource/eslint-plugin-sonarjs/blob/b8bf961931513d46a88a0d3bb48f79c04ecf4bd7/src/rules/cognitive-complexity.ts#L49C1-L50C1) type `(number | 'metric' | 'sonar-runtime')[]` does not seem to be correct if you look at the schema, which is:

```js
[
  { type: 'integer', minimum: 0 },
  { enum: ['sonar-runtime', 'metric'] }
]
```

If you compile this schema, you will get `[number, 'sonar-runtime' | 'metric']`, and you may want to have them optional so you could omit them.

Now, if your typings are used, this is valid from the typescript (types) perspective:
```js
{
  // 'error' from combining your typings with eslint Linter.RuleEntry
  'sonarjs/cognitive-complexity': ['error', 5, 'metric', 'metric', 5, 'metric', 'metric', 'sonar-runtime', 'sonar-runtime', 5, 9]
}
```

Of course, that does not break `create` method in any way because of [this](https://github.com/SonarSource/eslint-plugin-sonarjs/blob/b8bf961931513d46a88a0d3bb48f79c04ecf4bd7/src/rules/cognitive-complexity.ts#L71C1-L73C65) way of extracting the options. Anyways, it is odd. I was compiling schemas of rules for different ESLint plugins and found this inconsistency between types.

**Comment by ilia-kebets-sonarsource** [comment]
Thank you for the information, we will try to fix this in our next sprint.

**Comment by Wohops** [comment]
This issue has been migrated to Jira. [ESLINTJS-33](https://sonarsource.atlassian.net/browse/ESLINTJS-33)
